File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/interstitials/InterstitialZoneActivity.java
package com.applovin.enterprise.apps.demoapp.ads.applovin.interstitials;

import android.os.Bundle;
import android.widget.Button;

import com.applovin.adview.AppLovinInterstitialAd;
import com.applovin.adview.AppLovinInterstitialAdDialog;
import com.applovin.enterprise.apps.demoapp.R;
import com.applovin.enterprise.apps.demoapp.ui.BaseAdActivity;
import com.applovin.sdk.AppLovinAd;
import com.applovin.sdk.AppLovinAdClickListener;
import com.applovin.sdk.AppLovinAdDisplayListener;
import com.applovin.sdk.AppLovinAdLoadListener;
import com.applovin.sdk.AppLovinAdVideoPlaybackListener;
import com.applovin.sdk.AppLovinSdk;

/**
 * Created by suyashsaxena on 06/20/18.
 */
public class InterstitialZoneActivity
        extends BaseAdActivity
        implements AppLovinAdLoadListener, AppLovinAdDisplayListener, AppLovinAdClickListener, AppLovinAdVideoPlaybackListener
{
    private AppLovinAd                   currentAd;
    private AppLovinInterstitialAdDialog interstitialAd;
    private Button                       showButton;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate( savedInstanceState );
        setContentView( R.layout.activity_interstitial_manual_loading );

        setupCallbacksRecyclerView();

        interstitialAd = AppLovinInterstitialAd.create( AppLovinSdk.getInstance( this ), this );
        interstitialAd.setAdLoadListener( this );
        interstitialAd.setAdDisplayListener( this );
        interstitialAd.setAdClickListener( this );
        interstitialAd.setAdVideoPlaybackListener( this ); // This will only ever be used if you have video ads enabled.

        final Button loadButton = findViewById( R.id.loadButton );
        showButton = findViewById( R.id.showButton );

        loadButton.setOnClickListener( v -> {
            showButton.setEnabled( false );

            AppLovinSdk.getInstance( getApplicationContext() ).getAdService().loadNextAdForZoneId( "YOUR_ZONE_ID ", this );
        } );

        showButton.setOnClickListener( v -> {
            if ( currentAd != null )
            {
                interstitialAd.showAndRender( currentAd );
            }
        } );
    }

    //region Ad Load Listener

    @Override
    public void adReceived(AppLovinAd appLovinAd)
    {
        logCallback();
        currentAd = appLovinAd;
        showButton.setEnabled( true );
    }

    @Override
    public void failedToReceiveAd(int errorCode)
    {
        // Look at AppLovinErrorCodes.java for list of error codes
        logCallback();

        showButton.setEnabled( true );
    }

    //endregion

    //region Ad Display Listener

    @Override
    public void adDisplayed(AppLovinAd appLovinAd) { logCallback(); }

    @Override
    public void adHidden(AppLovinAd appLovinAd) { logCallback(); }

    //endregion

    //region Ad Click Listener

    @Override
    public void adClicked(AppLovinAd appLovinAd) { logCallback(); }

    //endregion

    //region Ad Video Playback Listener

    @Override
    public void videoPlaybackBegan(AppLovinAd appLovinAd) { logCallback(); }

    @Override
    public void videoPlaybackEnded(AppLovinAd appLovinAd, double percentViewed, boolean wasFullyViewed) { logCallback(); }

    //endregion
}

File: AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/applovin/interstitials/InterstitialZoneActivity.kt
package com.applovin.enterprise.apps.demoapp.ads.applovin.interstitials

import android.os.Bundle
import com.applovin.adview.AppLovinInterstitialAd
import com.applovin.enterprise.apps.demoapp.ads.AdStatusActivity

import com.applovin.sdk.AppLovinAd
import com.applovin.sdk.AppLovinAdDisplayListener
import com.applovin.sdk.AppLovinAdLoadListener
import com.applovin.sdk.AppLovinAdVideoPlaybackListener
import com.applovin.sdk.AppLovinSdk
import kotlinx.android.synthetic.main.activity_interstitial_manual_loading.*
import com.applovin.enterprise.apps.demoapp.R

class InterstitialZoneActivity : AdStatusActivity()
{
    private var currentAd: AppLovinAd? = null
    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_interstitial_manual_loading)

        adStatusTextView = status_label

        val sdk = AppLovinSdk.getInstance(applicationContext)
        val interstitialAdDialog = AppLovinInterstitialAd.create(sdk, this)

        loadButton.setOnClickListener {
            log("Interstitial loading...")
            showButton.isEnabled = false

            AppLovinSdk.getInstance(applicationContext).adService.loadNextAdForZoneId("YOUR_ZONE_ID", object : AppLovinAdLoadListener
            {
                override fun adReceived(ad: AppLovinAd)
                {
                    log("Interstitial Loaded")
                    currentAd = ad
                    runOnUiThread { showButton.isEnabled = true }
                }

                override fun failedToReceiveAd(errorCode: Int)
                {
                    // Look at AppLovinErrorCodes.java for list of error codes
                    log("Interstitial failed to load with error code " + errorCode)
                }
            })
        }

        showButton.setOnClickListener {
            currentAd?.let {
                interstitialAdDialog.showAndRender(it)
            }
        }

        //
        // Optional: Set ad display, ad click, and ad video playback callback listeners
        //
        interstitialAdDialog.setAdDisplayListener(object : AppLovinAdDisplayListener
                                                  {
                                                      override fun adDisplayed(appLovinAd: AppLovinAd)
                                                      {
                                                          log("Interstitial Displayed")
                                                      }

                                                      override fun adHidden(appLovinAd: AppLovinAd)
                                                      {
                                                          log("Interstitial Hidden")
                                                      }
                                                  })

        interstitialAdDialog.setAdClickListener { log("Interstitial Clicked") }

        // This will only ever be used if you have video ads enabled.
        interstitialAdDialog.setAdVideoPlaybackListener(object : AppLovinAdVideoPlaybackListener
                                                        {
                                                            override fun videoPlaybackBegan(appLovinAd: AppLovinAd)
                                                            {
                                                                log("Video Started")
                                                            }

                                                            override fun videoPlaybackEnded(appLovinAd: AppLovinAd, percentViewed: Double, wasFullyViewed: Boolean)
                                                            {
                                                                log("Video Ended")
                                                            }
                                                        })
    }
}
